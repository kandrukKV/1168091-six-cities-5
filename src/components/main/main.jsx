import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {changeCityAction, changeSortTypeAction} from "../../store/action";
import Header from "../header/header";
import CitiesList from "../cities-list/cities-list";
import OffersContent from "../offers-content/offers-content";
import OffersContentEmpty from "../offers-content-empty/offers-content-empty";
import {getCurrentCitySelector, getCitiesSelector, getCurrentSortTypeSelector, getFilteredOffersSelector} from "../../store/selectors";

const Main = (props) => {

  const {
    cards,
    currentCity,
    cities,
    currentSortType,
    onChangeCity,
    onChangeSortType
  } = props;

  return (
    <div className="page page--gray page--main">
      <Header/>
      <main className={`page__main page__main--index ${cards.length || `page__main--index-empty`}`}>
        <CitiesList
          cities={cities}
          currentCity={currentCity}
          onChangeCity={onChangeCity}
        />
        {
          cards.length ?
            <OffersContent
              cards={cards}
              currentCity={currentCity}
              currentSortType={currentSortType}
              onChangeSortType={onChangeSortType}
            /> :
            <OffersContentEmpty
              currentCity={currentCity}
            />
        }
      </main>
    </div>
  );

};

Main.propTypes = {
  cards: PropTypes.array.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentCity: PropTypes.string.isRequired,
  currentSortType: PropTypes.string.isRequired,
  onChangeCity: PropTypes.func.isRequired,
  onChangeSortType: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  cards: getFilteredOffersSelector(state),
  currentCity: getCurrentCitySelector(state),
  cities: getCitiesSelector(state),
  currentSortType: getCurrentSortTypeSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  onChangeCity(city) {
    dispatch(changeCityAction(city));
  },
  onChangeSortType(filterName) {
    dispatch(changeSortTypeAction(filterName));
  },
});

export {Main};
export default connect(mapStateToProps, mapDispatchToProps)(Main);
