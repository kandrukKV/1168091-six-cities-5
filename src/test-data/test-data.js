export const noon = () => {};

export const additions = [
  `addition1`,
  `addition2`,
  `addition3`,
  `addition4`,
  `addition5`,
];

export const cities = [
  `Amsterdam`,
  `Paris`,
];

export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: false,
    rating: 1,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.902976,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 2,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.388540000000006,
      longitude: 4.899976,
      zoom: 16
    },
    id: 2
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: true,
    rating: 3,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.364540000000005,
      longitude: 4.9019759999999994,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: false,
    rating: 4,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.886976,
      zoom: 16
    },
    id: 4
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 5,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16
    },
    id: 5
  }
];

export const mosckReviews = [
  {
    id: 1,
    user: {
      id: 18,
      name: `Sophie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`
    },
    rating: 3,
    comment: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`,
    date: `2020-11-02T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 16,
      name: `Mollie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`
    },
    rating: 2,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-11-02T13:38:44.752Z`
  }
];
