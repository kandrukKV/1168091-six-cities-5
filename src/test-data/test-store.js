import configureStore from "redux-mock-store";
import {mockOffers, mockReviews, cities} from "../test-data/test-data";

const createStore = configureStore();

const mockStore = createStore({
  API_DATA: {
    offers: mockOffers,
    favoriteOffers: mockOffers,
    offerDetails: {
      card: mockOffers[0],
      reviews: mockReviews,
      nearPlaces: mockOffers
    }
  },
  APP_STATE: {
    currentCity: `Amsterdam`,
    cities,
    currentSortType: `Popular`,
    activeCard: null,
    reviewFormState: `EDITING`
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    authInfo: null
  }
});

export {mockStore};
