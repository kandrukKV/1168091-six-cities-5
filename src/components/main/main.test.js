import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import {Main} from "./main";
import {cities, noon, mockOffers} from "../../test-data/test-data";

it(`Main is rendered correctly with offers`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <Main
        cities={cities}
        cards={mockOffers}
        currentCity={mockOffers[0].city.name}
        onChangeCity={noon}
        currentSortType={`sortType`}
        onChangeSortType={noon}
      />
  );
  expect(tree).toMatchSnapshot();
});

it(`Main is rendered correctly with out offers`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <Main
        cities={cities}
        cards={[]}
        currentCity={mockOffers[0].city.name}
        onChangeCity={noon}
        currentSortType={`sortType`}
        onChangeSortType={noon}
      />
  );
  expect(tree).toMatchSnapshot();
});
