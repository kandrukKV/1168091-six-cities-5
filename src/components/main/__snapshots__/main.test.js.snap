// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main is rendered correctly with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--index 5"
  >
    <CitiesList
      cities={
        Array [
          "Amsterdam",
          "Paris",
        ]
      }
      currentCity="Amsterdam"
      onChangeCity={[Function]}
    />
    <OffersContent
      cards={
        Array [
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.385540000000006,
              "longitude": 4.902976,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 1,
            "title": "Perfectly located Castro",
            "type": "house",
          },
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 2,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": true,
            "isPremium": true,
            "location": Object {
              "latitude": 52.388540000000006,
              "longitude": 4.899976,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 2,
            "title": "Perfectly located Castro",
            "type": "house",
          },
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 3,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": false,
            "isPremium": true,
            "location": Object {
              "latitude": 52.364540000000005,
              "longitude": 4.9019759999999994,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 3,
            "title": "Perfectly located Castro",
            "type": "house",
          },
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 4,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": true,
            "isPremium": false,
            "location": Object {
              "latitude": 52.385540000000006,
              "longitude": 4.886976,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 4,
            "title": "Perfectly located Castro",
            "type": "house",
          },
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 5,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": true,
            "isPremium": true,
            "location": Object {
              "latitude": 52.397540000000006,
              "longitude": 4.9099759999999995,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 5,
            "title": "Perfectly located Castro",
            "type": "house",
          },
        ]
      }
      currentCity="Amsterdam"
      currentSortType="sortType"
      onChangeSortType={[Function]}
    />
  </main>
</div>
`;

exports[`Main is rendered correctly with out offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <CitiesList
      cities={
        Array [
          "Amsterdam",
          "Paris",
        ]
      }
      currentCity="Amsterdam"
      onChangeCity={[Function]}
    />
    <OffersContentEmpty
      currentCity="Amsterdam"
    />
  </main>
</div>
`;
